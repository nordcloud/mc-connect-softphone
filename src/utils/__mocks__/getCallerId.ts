export const getCallerId = jest.fn().mockResolvedValue('dummyCallerId');
