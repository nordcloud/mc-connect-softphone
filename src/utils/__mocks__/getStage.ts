export const getStage = jest.fn().mockReturnValue('prod');
