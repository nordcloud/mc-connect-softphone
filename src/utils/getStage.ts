import { Stage } from '../types';
import { assertStageName } from './assertStageName';

export function getStage(): Stage {
  const { STAGE } = process.env;

  assertStageName(STAGE);

  return STAGE;
}
