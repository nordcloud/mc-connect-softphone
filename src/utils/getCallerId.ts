import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
import { DynamoDBDocumentClient, GetCommand } from '@aws-sdk/lib-dynamodb';

import { CALLER_ID_TABLE } from '../consts';
import { getStage } from './getStage';

const ddbClient = new DynamoDBClient({});
const ddbDocClient = DynamoDBDocumentClient.from(ddbClient);

export async function getCallerId(username: string) {
  if (getStage() === 'local') {
    return 'mockedCallerId';
  }

  const getCommand = new GetCommand({
    TableName: CALLER_ID_TABLE,
    Key: { username },
  });

  const { Item } = await ddbDocClient.send(getCommand);

  return Item?.callerId;
}
