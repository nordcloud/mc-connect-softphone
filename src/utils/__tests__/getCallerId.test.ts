import { DynamoDBDocumentClient, GetCommand } from '@aws-sdk/lib-dynamodb';
import { mockClient } from 'aws-sdk-client-mock';

import { getCallerId } from '../getCallerId';
import { getStage } from '../getStage';

jest.mock('../getStage');

const ddbMock = mockClient(DynamoDBDocumentClient);

const args = ['username'] as const;

beforeEach(() => {
  ddbMock.reset();
  ddbMock.on(GetCommand).resolves({ Item: { callerId: 'dummyCallerId' } });
});

describe('getCallerId', () => {
  it('sends "GetCommand" with correct args', async () => {
    await getCallerId(...args);
    expect(ddbMock.commandCalls(GetCommand)[0].args[0].input).toMatchSnapshot();
  });

  it('resolves with a correct value', () => {
    return expect(getCallerId(...args)).resolves.toMatchSnapshot();
  });

  it('resolves with a correct value when output from GetCommand is empty', () => {
    ddbMock.on(GetCommand).resolvesOnce({});
    return expect(getCallerId(...args)).resolves.toMatchSnapshot();
  });

  it('resolves with a correct value when running in "local" stage', () => {
    jest.mocked(getStage).mockReturnValueOnce('local');
    expect(getCallerId(...args)).resolves.toMatchSnapshot();
  });
});
