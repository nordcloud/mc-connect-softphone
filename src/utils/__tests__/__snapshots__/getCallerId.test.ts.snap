// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCallerId resolves with a correct value 1`] = `"dummyCallerId"`;

exports[`getCallerId resolves with a correct value when output from GetCommand is empty 1`] = `undefined`;

exports[`getCallerId resolves with a correct value when running in "local" stage 1`] = `"mockedCallerId"`;

exports[`getCallerId sends "GetCommand" with correct args 1`] = `
{
  "Key": {
    "username": "username",
  },
  "TableName": "mc-connect-softphone-prod-agents-caller-id",
}
`;
